### PR TITLE
Adding backwards compatibility test for ml-commons plugin

### PR DESCRIPTION
**Pull request by b4sjoo**
Signed-off-by: Sicheng Song <sicheng.song@outlook.com>

### Description
This PR should add backwards compatibility test for our ml-commons plugin in order to test the backwards compatibility of the latest minor version of the previous major version of our plugin.
 
### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
- [ ] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -5,4 +5,5 @@ client/build/
 common/build/
 ml-algorithms/build/
 plugin/build/
+plugin/src/test/resources/bwc/*
```
Comment (ylwu-amzn) [inline .gitignore:8]: Anywhere else using this path pattern? I don't see this in other places.

File: plugin/build.gradle [hunk 2]
```diff
@@ -369,3 +369,166 @@ tasks.withType(licenseHeaders.class) {
 checkstyle {
     toolVersion = '8.29'
 }
+
+String bwcVersion = "1.3.0.0"
```
Comment (ylwu-amzn) [inline plugin/build.gradle:373]: Add how to run bwc test to [developer guide](https://github.com/opensearch-project/ml-commons/blob/main/DEVELOPER_GUIDE.md) like AD https://github.com/opensearch-project/anomaly-detection/blob/main/DEVELOPER_GUIDE.md

**Comment by ylwu-amzn** [comment]
I think we should add Github workflow file like AD https://github.com/opensearch-project/anomaly-detection/blob/main/.github/workflows/test_bwc.yml. So for any PR or code push, it will trigger bwc test automatically.

File: .gitignore (hunk 1 above)
Comment (b4sjoo) [reply .gitignore:8]: This path is for storing the downloaded old and updated version of ml-commons plugins used in bwc test

**Review by rbhavna — APPROVED** [review]

**Review by ylwu-amzn — APPROVED** [review]

**Pull request merged**